doc/proposals/scorecard-custom-tests.md

```diff
@@ -0,0 +1,403 @@
+To support this type of test, test logic would be stored within a container image, then the test  essentially becomes the execution of that test container image.  You would then create a test assertion that checks the test pod’s status.   The scorecard would read the Complex test Pod’s log for more detailed information that could be presented to the end-user.
+
+In this scenario, a test setup step would look as follows:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: custom-test-of-something
+  namespace: mynamespace
+spec:
+  restartPolicy: OnFailure
+  containers:
+  - name: mytest
+    image: someuser/sometestimage:3.6.9-ubi8
+    Env:
+…
+```
+
+Then the test assertion might look as follows:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+ name: custom-test-of-something
+status:
+ phase: Successful
+```
+
+#### Static Check Test Format
+
+Custom tests that are merely a static check are implemented in a similar fashion as a Complex test, however, they do not require a Kube connection necessarily.  A static check custom test would be started as a Pod, with any required on-disk manifests bundled within a ConfigMap that is available to the static check test Pod.  A test assertion would be used to determine if the static check Pod was successful or not just like the other custom test types.  Static check test logs are contained within the Pod’s log, the scorecard would read the Pod’s logs to display back to the end user.
+
+### Custom Test Assertion Output Format
+
+With all scorecard test types, the test assertion output is translated into a scorecard v1alpha2 ScorecardTestResult structure for consistency in reporting test results.
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, the test assertions are YAML files, test steps are YAML files as well.  Container images holding the custom test logic are expected to be pulled from a registry that is accessible in the Kube test environment that is executing the scorecard tests.
+
+Custom tests and the YAML files that make them up would be accessible via URL or from a local disk location.  Custom test locations would be configured from within the scorecard configuration file.  For example, the scorecard configuration might be as follows to indicate the presence of custom tests from different sources:
+```
+scorecard:
+  # Setting a global scorecard option
+  output: json
+  plugins:
+    - tests:
+        init-timeout: 60
+        test-paths:
+          - "tests/scorecard-tests"
+          - "https://some-of-my-tests"
+          - "tests/mycustom-tests
+cr-manifest:
+          - "deploy/crds/cache.example.com_v1alpha1_memcached_cr.yaml"
+          - "deploy/crds/cache.example.com_v1alpha1_memcached_cr2.yaml"
+  …
+```
+
+
+Here is an on-disk representation of a set of custom tests:
+```
+tests
+├── e2e
+│   └── example-test
+│       ├── 00-assert.yaml
```

Comment:
Can we have those filenames including paths be reflected in the code examples for static, assertion and complex tests above? So people understand where stuff ends up.

Reply:
I updated the samples to show which files match up and what each file does, hopefully it is clearer now.